src/gasman.c

```diff
@@ -1955,7 +1955,16 @@ UInt CollectBags (
         // extract the head from the linked list
         first = MarkedBags;
         MarkedBags = LINK_BAG(first);
-        LINK_BAG(first) = MARKED_ALIVE(first);
+        // gasman is inconsistent if a bag starting at YoungBags is
```

Comment:
Ah, maybe what is meant here is that in some places, Gasman will treat a bag with `PTR_BAG(bag) == YoungBags` as young, and in other as old? Shouldn't we then just fix it one way or another? (Maybe that's too big a change for this PR, I didn't look at the code, just wondering)

Reply:
My problem was you can't actually really check you are right, as that condition is never true :)

Reply:
I rephrased this comment, and also moved it to a seperate check.

Basically, I spend a while worrying about PTR_BAG(bag) == YoungBags, then decided I'm sure it can't ever happen. However, just for sanity's sake I would like to add a check, just to make sure my reasoning was right (if this assert ever triggers, we'll have to find out why, and then check bags starting at YoungBags carefully).